test/items-updater-test.js

```diff
@@ -0,0 +1,75 @@
+/* global describe it before after */
+
+const sinon = require('sinon')
+
+const assert = require('assert')
+const ItemsUpdater = require('./../lib/items-updater')
+const db = require('./../lib/db')
+const ItemSierraRecord = require('./../lib/models/item-sierra-record')
+
+describe('Items Updater', function () {
+  describe('_getBibIdsForItemStatements', function () {
+    before(() => {
+      sinon.stub(db, 'getStatement').callsFake((type, subjectId, predicate) => {
+        // Mocked for single test lookup:
+        return Promise.resolve({
+          subject_id: 'ci5651186',
+          predicate: 'nypl:bnum',
+          object_id: 'urn:bnum:cb4194299'
+        })
+      })
+    })
+
+    after(() => {
+      db.getStatement.restore()
+    })
+
+    it('should derive bibId from statements if given', function () {
+      let itemRecord = ItemSierraRecord.from({
+        'id': '5651186',
+        'nyplSource': 'recap-cul',
+        'nyplType': 'item',
+        'bibIds': ['1234-bib-id-given-in-record']
+      })
+      // First run this record through the serializer to extract item statements
+      return ItemsUpdater.prototype.extractStatements(itemRecord)
+        .then((statements) => {
+          // Now run those statements through this utility function that
+          // identifies the bibIds that need to be reindexed based on the
+          // item statements:
+          return ItemsUpdater.prototype._getBibIdsForItemStatements(statements)
+            .then((bibIdsToReindex) => {
+              // We expect a single bibId to be extracted matching the value
+              // in the moch marc-json (with 'cb' prefix):
+              assert.equal(bibIdsToReindex.length, 1)
+              assert.equal(bibIdsToReindex[0], 'cb1234-bib-id-given-in-record')
+            })
+        })
+    })
+
```

Comment:
Let's talk with @jobinthomasnypl .

I'm not sure if the retrievers will publish messages that are totally missing a `bibIds` property or
if they'd be an empty Array.

Reply:
Mostly talking to myself here...I just tested this locally by adding `'bibIds': []` to the fixture and it still passes, which is good - just to confirm with @nonword - the reason why it passes is because...

`let bibIdStatement = subjectIdGroups[subjectId].filter((statement) => statement.predicate === 'nypl:bnum')[0]`

on [this line of items-updater.js](https://github.com/NYPL-discovery/pcdm-store-updater/pull/54/files#diff-edaae7b6abb00c2e8a1e6d605b28daa3R162) will pretty much be calling `[][0]` (element zero of an empty array).

Reply:
That's correct. The `_getBibIdsForItemStatements` function that does most of the work here operates on the result of the serialization, not on the marc-in-json document directly. So it doesn't matter what form an absent bibIds property takes; If the serialization process fails to extract bibIds (either because the property is `undefined`, `null`, or `[]`), `_getBibIdsForItemStatements` will look up the bibId.